cmd/start.go

```diff
@@ -233,3 +236,49 @@ func initializeMetrics(
 		netProvider,
 	)
 }
+
+func initializeDiagnostics(
+	ctx context.Context,
+	config *config.Config,
+	netProvider net.Provider,
+	stakeMonitor chain.StakeMonitor,
+) {
+	registry, isConfigured := diagnostics.Initialize(
+		config.Diagnostics.Port,
+	)
+	if !isConfigured {
+		logger.Infof("diagnostics are not configured")
+		return
+	}
+
+	registry.RegisterSource("peers", func() string {
+		connectedPeers := netProvider.ConnectionManager().ConnectedPeers()
+
+		peersList := make([]map[string]interface{}, len(connectedPeers))
+		for i := 0; i < len(connectedPeers); i++ {
+			peerAddress, err := netProvider.ConnectionManager().GetPeerPublicKey(connectedPeers[i])
+			if err == nil {
+				hasMinimumStake := false
+				hasMinimumStake, _ = stakeMonitor.HasMinimumStake(peerAddress.X.String())
```

Comment:
What's the point of checking the minimum stake here? If we have a peer on the connected peers list we can assume they had the minimum stake because the handshake during the connection attempt was successful. Of course, it's possible the peer just loosed the stake and will be disconnected by the firewall soon. However, I think the only question answered by this diagnostic source should be "what specific peers my client is connected to at this point?"

Reply:
Well, yeah... Actually I hoped to raise discussion about other useful information which could be collected as the part of 'connected_peers' diagnostics source, but as @pdyraga suggested - I'll rather do it in separate issue. Deleting for now.